### PR TITLE
Build research station

### PR DESCRIPTION
**Pull request by schai6**
Connected to #21 

Build a research station when clicking the build action.
Decrement remainingResearchStation count when a research station is built.
Refactored Board to use firestore.

### CODE REVIEW

File: src/components/ActionFooter.js [hunk 1]
```diff
@@ -45,8 +45,28 @@ const mapStateToProps = (state) => {
...
+      if (!currentResearchStation && remainingResearchStations >= 0) {
+        currentCityRef.update({ researchStation: true });
+        remainingResearchStations--;
+        game.ref.update({ remainingResearchStations });
+      } else if (currentResearchStation) {
+        console.log('There is already a research station at the current city');
+      } else {
+        console.log('You lose because no more research stations');
+      }
+    });
```
Comment (sarahzhao25) [inline src/components/ActionFooter.js]: Add error handling for this promise. in ActionFooter.js

File: src/components/CityMarkers.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+});
+
+const iconContainer = {
+  redIcon: new Icon({ iconUrl: 'https://lh5.ggpht.com/JUGn9I-kMM3LriNMpdUA6Z1_NZksTHCndCJ7SqSG0CkF6P-rBHUS91_aAiWfNpKSoQ=w300' }),
+  blueIcon: new Icon({ iconUrl: 'http://lobelpost.com/v17/files/stacks-image-12a7505.png' }),
+  yellowIcon: new Icon({ iconUrl: 'https://lh3.googleusercontent.com/nmIfOaurHcAvlxd6OksvTYkF1thhsEnpxV2x0PvJ8zTxS-uAX0r7BWQxM20XTL6SrQ' }),
+  blackIcon: new Icon({ iconUrl: 'https://totalsororitymove.com/wp-content/uploads/user_avatars/blackball.png' })
+};
+
+const CityMarkers = ({ cities } ) => {
```
Comment (sarahzhao25) [inline src/components/CityMarkers.js]: Delete the space after } and before )

File: src/components/PlayerMarkers.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  'Contingency Planner': new PlayerIcon({iconUrl: 'assets/images/cont_planner.png'}),
+  Dispatcher: new PlayerIcon({iconUrl: 'assets/images/dispatcher.png'}),
+  Medic: new PlayerIcon({iconUrl: 'assets/images/medic.png'}),
+  'Operations Expert': new PlayerIcon({iconUrl: 'assets/images/ops_expert.png'}),
+  'Quarantine Specialist': new PlayerIcon({iconUrl: 'assets/images/quar_spec.png'}),
+  Researcher: new PlayerIcon({iconUrl: 'assets/images/researcher.png'}),
+  Scientist: new PlayerIcon({iconUrl: 'assets/images/scientist.png'}),
+};
+
+const PlayerMarkers = ({ cities, players } ) => {
```
Comment (sarahzhao25) [inline src/components/PlayerMarkers.js]: teehee, here too. these are probably mine aren't they...

File: src/components/ActionFooter.js [hunk 4]
```diff
@@ -45,8 +45,28 @@ const mapStateToProps = (state) => {
...
+};
+
+function setCityResearchStation(firestore, currentTurn, cities) {
+  Promise.all([currentTurn && currentTurn.get(), firestore.get(`games/${doc}`)])
+    .then(([player, game]) => {
+      const currentCity = player.data().currentCity;
+      const currentCityRef = game.ref.collection('cities').doc(currentCity);
+      const currentResearchStation = cities[currentCity].researchStation;
+      let remainingResearchStations = game.data().remainingResearchStations;
+      if (!currentResearchStation && remainingResearchStations >= 0) {
```
Comment (sarahzhao25) [inline src/components/ActionFooter.js]: Shouldn't this be remainingResearchStations > 0?

**Comment by schai6** [comment]
Fixed!

**Pull request merged**